Add CHANGELOG for 1.0.0 (#4858)
holoviz/panel: CHANGELOG.md

### Major Features

The three main features we want to highlight as part of this release are:

- Add new and highly performant layout engine, addressing many performance and layout issues ([#4326](https://github.com/holoviz/panel/pull/4326), [#4463](https://github.com/holoviz/panel/pull/4463), [#4491](https://github.com/holoviz/panel/pull/4491), [#4503](https://github.com/holoviz/panel/pull/4503), [#4522](https://github.com/holoviz/panel/pull/4522), [#4690](https://github.com/holoviz/panel/pull/4690))
- Add unified Design and Theme system ([#4413](https://github.com/holoviz/panel/pull/4413), [#4475](https://github.com/holoviz/panel/pull/4475), [#4466](https://github.com/holoviz/panel/pull/4466), [#4540](https://github.com/holoviz/panel/pull/4540), [#4548](https://github.com/holoviz/panel/pull/4548), [#4770](https://github.com/holoviz/panel/pull/4770), [#4792](https://github.com/holoviz/panel/pull/4792), [#4793](https://github.com/holoviz/panel/pull/4793), [#4801](https://github.com/holoviz/panel/pull/4801))

> - Implement support for passing parameters, widgets and bound functions by reference ([#4495](https://github.com/holoviz/panel/pull/4495), [#4505](https://github.com/holoviz/panel/pull/4505), [#4603](https://github.com/holoviz/panel/pull/4603), [#4606](https://github.com/holoviz/panel/pull/4606))

Reviewer: Maybe explain that this approach supports reactive programming and is much simpler than alternatives that require explicit event handling?

Reviewer: ```suggestion
- Implement support for passing parameters, widgets and bound functions by reference to support an easy-to-use reactive-programming model ([#4495](https://github.com/holoviz/panel/pull/4495), [#4505](https://github.com/holoviz/panel/pull/4505), [#4603](https://github.com/holoviz/panel/pull/4603), [#4606](https://github.com/holoviz/panel/pull/4606))
```

> Some of the changes related to the new layout engine and design system will require some updates to your existing applications. To make this transition as smooth as possible we also added an [upgrade/migration guide](https://panel.holoviz.org/panel/upgrade.html).

Reviewer: ```suggestion
Some of the changes related to the new layout engine and design system will require small updates to your existing applications. To make this transition as smooth as possible we have provided an [upgrade/migration guide](https://panel.holoviz.org/panel/upgrade.html).
```

### New Components

This release also adds a number of exciting new components you can leverage in your applications:

- Add `Swipe` layout ([#3007](https://github.com/holoviz/panel/pull/3007))
- Add `Switch` widget ([[#4130](https://github.com/holoviz/panel/pull/4130)](https://github.com/holoviz/panel/pull/4130))
- Add `Vizzu` pane ([#4226](https://github.com/holoviz/panel/pull/4226), [#4739](https://github.com/holoviz/panel/pull/4739))
- Add `BrowserInfo` model to expose browser window and navigator APIs ([#4533](https://github.com/holoviz/panel/pull/4533))
- Add `BasicAuth` provider for quick password based auth ([#4684](https://github.com/holoviz/panel/pull/4684))
- Add `FloatPanel` layout ([#4707](https://github.com/holoviz/panel/pull/4707), [#4711](https://github.com/holoviz/panel/pull/4711))
- Add a `SlidesTemplate` based on reveal.js to create interactive presentations ([#4798](https://github.com/holoviz/panel/pull/4798))

### Major enhancements

There are also a number of major enghancements in this release that we are very excited about:

> - Improved markdown rendering ([#4688](https://github.com/holoviz/panel/pull/4688))

Reviewer: ```suggestion
- Improved Markdown rendering ([#4688](https://github.com/holoviz/panel/pull/4688))
```

- Add support for tooltips on widgets ([[#4130](https://github.com/holoviz/panel/pull/4130)](https://github.com/holoviz/panel/pull/4130), [#4621](https://github.com/holoviz/panel/pull/4621), [#4643](https://github.com/holoviz/panel/pull/4643))
- Ensure `.ipynb` and `.md` based apps can be used as `--index` ([#4432](https://github.com/holoviz/panel/pull/4432))

> - Implement Matplotlib format support and responsive Image sizing ([#4514](https://github.com/holoviz/panel/pull/4514))

Reviewer: ```suggestion
- Add support for selecting `format` and `encoding` for Matplotlib image output and implemented  responsive Image sizing ([#4514](https://github.com/holoviz/panel/pull/4514))
```

- Add support for icons on `Button` ([#4797](https://github.com/holoviz/panel/pull/4797))

> - Add generic `Image` pane ([#4551](https://github.com/holoviz/panel/pull/4551))

Reviewer: ```suggestion
- Add generic `Image` pane that auto-detects the image filetype ([#4551](https://github.com/holoviz/panel/pull/4551))
```

- Add support for writing applications in Markdown ([#4602](https://github.com/holoviz/panel/pull/4602))
- Improve support for inline resources for Jupyter ([#3013](https://github.com/holoviz/panel/pull/3013), [#4787](https://github.com/holoviz/panel/pull/4787))
- Add ability to reuse sessions to speed up rendering ([#3679](https://github.com/holoviz/panel/pull/3679), [#4658](https://github.com/holoviz/panel/pull/4658))
- Improve notebook resource and extension loading ([#4752](https://github.com/holoviz/panel/pull/4752))
- Add ability to add global loading spinner to application(s) ([#4659](https://github.com/holoviz/panel/pull/4659))

### Documentation

> The last major change we want to point out is a complete overhaul of the documentation, moving from long and difficult to navigate user guides to distinct how-to guides and longer explanation sections. We also put in significant effort to ensure that most of our documentation can be run interactively in Pyodide or JupyterLite.

Reviewer: ```suggestion
The last major change we want to highlight is a complete overhaul of the documentation, moving from long and difficult-to-navigate user guides to distinct easily applied how-to guides along with separate, longer explanation sections. We also put in significant effort to ensure that most of our documentation can be run interactively in Pyodide or JupyterLite.
```

- Upgrade Echarts to 5.4.1 ([#4538](https://github.com/holoviz/panel/pull/4538))
- Upgrade pyodide (0.23.1) and pyscript versions ([#4344](https://github.com/holoviz/panel/pull/4344))
- Add support for altair and vega-lite v5 ([#4488](https://github.com/holoviz/panel/pull/4488))
- Add support for latest versions of ipywidgets ([#4716](https://github.com/holoviz/panel/pull/4716), [#4766](https://github.com/holoviz/panel/pull/4766), [#4779](https://github.com/holoviz/panel/pull/4779))

#### Deprecations

- Deprecate `IDOM` pane ([[#4293](https://github.com/holoviz/panel/pull/4293)](https://github.com/holoviz/panel/pull/4293), [#4323](https://github.com/holoviz/panel/pull/4323))
- Deprecate `Viewable.app` ([[#4293](https://github.com/holoviz/panel/pull/4293)](https://github.com/holoviz/panel/pull/4293))
- Deprecate `Viewable.pprint` ([#4347](https://github.com/holoviz/panel/pull/4347))
- Deprecate and remove `RGGPlot`
- Rename `Ace` to `CodeEditor` ([#4627](https://github.com/holoviz/panel/pull/4627))

#### API changes & Backward Compatibility

> - Pandas becomes a (lazy) runtime dependency ([#4411](https://github.com/holoviz/panel/pull/4411))

Reviewer: ```suggestion
- Pandas is now only a (lazy) runtime dependency ([#4411](https://github.com/holoviz/panel/pull/4411))
```

- `Tabulator.frozen_rows` now respects the order of row in the data instead of the order in which the `frozen_rows` were defined ([[#4482](https://github.com/holoviz/panel/pull/4482)](https://github.com/holoviz/panel/pull/4482))
- Make `margin` defaults consistent across widgets and panes ([#4528](https://github.com/holoviz/panel/pull/4528))
- Extension calls must specify all required extensions ([#4562](https://github.com/holoviz/panel/pull/4562))
- The `.embed` method now returns a Mimebundle object for rendering ([#4791](https://github.com/holoviz/panel/pull/4791))
- Remove `panel examples` CLI command and pyct dependency ([#4691](https://github.com/holoviz/panel/pull/4691))
- Expose all layout components in top-level API ([#4696](https://github.com/holoviz/panel/pull/4696))

### Other Enhancements

#### Configuration

- Allow to set the log level of the Admin logger ([#3495](https://github.com/holoviz/panel/pull/3495))

> - Add `pn.state.served` to determine to simplify whether script is executed as application ([#4252](https://github.com/holoviz/panel/pull/4252))

Reviewer: ```suggestion
- Add `pn.state.served` to simplify determining whether script is executed as an application or in an interactive session ([#4252](https://github.com/holoviz/panel/pull/4252))
```

#### Miscelleanous

- Handle cancelling and empty value edit events on `Tabulator` ([#4343](https://github.com/holoviz/panel/pull/4343))
- Add favicon to base template ([#4626](https://github.com/holoviz/panel/pull/4626))
- Ensure `CrossSelector` filters apply on each keystroke ([#4339](https://github.com/holoviz/panel/pull/4339))
- Do not re-create `Vega.selections` object unless selections changed ([#4497](https://github.com/holoviz/panel/pull/4497))
- Standardize parameter mapping APIs ([#4386](https://github.com/holoviz/panel/pull/4386))
- Add `Plotly.link_figure` parameter ([#4333](https://github.com/holoviz/panel/pull/4333))
- Add jpeg support for `JPG` pane ([#4532](https://github.com/holoviz/panel/pull/4532))
- Make periodic callback `counter` a parameter ([#4134](https://github.com/holoviz/panel/pull/4134))
- Adding Echarts events ([#2174](https://github.com/holoviz/panel/pull/2174))
- Additional cache support ([#4663](https://github.com/holoviz/panel/pull/4663), [#4667](https://github.com/holoviz/panel/pull/4667))

### Bug fixes

- Fix caching on undecorated Parameterized method ([#4332](https://github.com/holoviz/panel/pull/4332))

> - Ensures global notification object can be used inside notebook callbacks ([#4331](https://github.com/holoviz/panel/pull/4331))

Reviewer: ```suggestion
- Ensure that global notification object can be used inside notebook callbacks ([#4331](https://github.com/holoviz/panel/pull/4331))
```
